presto-main/src/main/resources/webapp/assets/query.js

```diff
@@ -893,10 +893,30 @@ var QueryDetail = React.createClass({
     renderFailureInfo: function() {
         var query = this.state.query;
         if (query.failureInfo) {
+            var taskInfo = "N/A";
+            if (query.failureInfo.task != null) {
+                var task = query.failureInfo.task;
+                var tasks = this.getTasksFromStage(this.state.lastSnapshotTasks)
+                                .filter(function(task) { return this.state.taskFilter(task.taskStatus.state) }, this);
+                if(tasks && task.length != 0) {
+                    var taskUri = tasks.reduce(function(prevResult,thisTask)  {
+                        if(thisTask.taskStatus.taskId == task) return thisTask.taskStatus.self;
+                        else return prevResult;
+                       },"");
+                    taskInfo = (
+                          <a href={ taskUri + "?pretty" }>
+                                { getTaskIdSuffix(task)}
+                          </a>
+                        );
+                } else {
+                    taskInfo = getTaskIdSuffix(task);
```

Comment:
Sorry, I don't understand. Under what conditions would there *not* be a URL for a task ID?